### DISCUSSION TITLE
loading grblhal on a BTT PICO for a laser engraver

### DISCUSSION THREAD

**Discussion by brandonsherring** [grblHAL/RP2040#96] [Q&A]
I am currently trying to build a laser engraver on a BTT PICO, I also tried building it on a BTT mini e3 v2 but have been having trouble getting it to compile on the e3v2, used web builder for the firmware for the btt pico and it came up just fine except for the way the e-stops and what not are mapped out.  could someone try and explain it to me, I mean how it is supposed to be wired for the hold, e-stop, cycle start are all using E0 stepper connector.  I am trying to get this going as soon as possible Have an idea for my daughters wedding I want to hopefully get done.
https://github.com/grblHAL/RP2040/blob/master/boards/btt_skr_pico_10_map.h

**Comment by terjeio** [comment]
I have just committed a [revised board map](https://github.com/grblHAL/RP2040/blob/master/boards/btt_skr_pico_10_map.h) that does not add feed hold and cycle start inputs, reset/estop will be mapped to the E0 limit input for 3-axis/non auto-squared configurations. Hope this helps.

**Comment by brandonsherring** [comment]
I"ve got my btt pico wotking with one exception, no pwm output.  either i am not understanding the path and how to wire it or it just isn't working.  there are 2 connection spots called IO20 fan1 and laser I tried both to 24v, both to ground  was running m3 s255 using iosender during test.

**Reply by terjeio** [reply]
There are two options, either P12 (Laser) which is 0-3.3V via a 100R resistor or P17 (Fan 3) which is an _open drain output_. If using P17 a pullup resistor to VCC (12 or 24V) or the voltage of your choice may be needed.

**Reply by brandonsherring** [reply]
I bought the laser tree lt-4lds which came with another circuit board which should work for the voltage its labeled driver adapter.  P-DA-01.  I had 24v and ground going to it with a single line from the laser and pin fan1 and neither worked I will try pin 17 fan 3 now.  Appreciate the help.    I think i figured it out, I had the fan hooked up pwm to ground not neg fan to pwm and 24v to pos fan...
